### PR TITLE
contractor baton cyborg stunning now uses left/right clicks, not combat mode

### LINKED ISSUE #58905: contractor batons only stun cyborgs if you're NOT on combat mode
this should probably be changed to match up with the left click/right click system that baton attacks against humans use to determine whether your target should be normal batonned or harmbatonned.

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

Contractor batons will now stun cyborgs on a left click and harmbaton them on a right click, instead of stunning cyborgs if you're NOT in combat mode and harmbatonning them if you are.

Stunning a cyborg with a contractor baton now puts your contractor baton on cooldown, just like knocking a human down does. Don't worry, you can still easily chainstun/baton 'n' bash borgs with contractor batons.

Attempting to stun a cyborg with a classic baton or a telebaton will play a CLONGing noise, produce a special message, and put your baton on cooldown. Harmbatonning cyborgs with classic batons and telebatons still works, of course.

The code for classic batons, telebatons, and contractor batons has been doc'd slightly better and refactored a bit.

Some edge cases involving cyborgs who somehow both are clumsy and can wield a classic baton (or a telebaton, a contractor baton, etc.) have been accounted for. This shouldn't affect normal gameplay unless an admin tries some REALLY weird stuff.

None of the above changes have affected stun batons, as they aren't classic baton subtypes.

This PR fixes https://github.com/tgstation/tgstation/issues/58905.

## Why It's Good For The Game

Contactor batons should determine whether they should stun or harmbaton a cyborg the same way that they determine whether they should knockdown or harmbaton a human. You also shouldn't need to turn off combat mode to stun a cyborg with a contractor baton, which is a pretty hostile action.

I have tested this PR, and the only oddity I've found is that attempting to stun a cyborg with a baton while it's on cooldown still produces sparks. This is a general issue with most objects that have nonzero force values and have a non-stamina damage type but also have a non-harmful effect when used on a cyborg, and I don't quite know how I would (safely) go about fixing it.

## Changelog
:cl: ATHATH
qol: Contractor batons will now stun cyborgs on a left click and harmbaton them on a right click, instead of stunning cyborgs if you're NOT in combat mode and harmbatonning them if you are.
fix: Stunning a cyborg with a contractor baton now puts your contractor baton on cooldown, just like knocking a human down does. Don't worry, you can still easily chainstun/baton 'n' bash borgs with contractor batons.
soundadd: Attempting to stun a cyborg with a classic baton or a telebaton will play a CLONGing noise, produce a special message, and put your baton on cooldown. Harmbatonning cyborgs with classic batons and telebatons still works, of course.
refactor: The code for classic batons, telebatons, and contractor batons has been doc'd slightly better and refactored a bit.
fix: Some edge cases involving cyborgs who somehow both are clumsy and can wield a classic baton have been accounted for. This shouldn't affect normal gameplay unless an admin tries some REALLY weird stuff. Like, clumsy contractor cyborg level weird.
/:cl:

### CODE REVIEW

**Comment by Sparkezel** [comment]
Balance change :trollface:

**Comment by ATH1909** [comment]
> Balance change :trollface:. Weren't batons meant codewise to stun borgs too iirc? Or just steal the power from the batong? I think there was an early return somewhere in batong code....

????

Stun batons are not affected by this change at all.

**Comment by maxymax13** [comment]
> 
> 
> Balance change :trollface:. Weren't batons meant codewise to stun borgs too iirc? Or just steal the power from the batong? I think there was an early return somewhere in batong code....

???????????

**Comment by LemonInTheDark** [comment]
image
Why?

**Comment by ATH1909** [comment]
> image
> Why?

The 18 second hardstun was absolutely hideous. I can remove that change if you want, though.

**Comment by LemonInTheDark** [comment]
I'd like that yeah

**Comment by ATH1909** [comment]
> I'd like that yeah

The absurdly long hardstun that someone probably just forgot to change when batons were changed to use stamina damage instead of hardstuns is back, as per your request.

**Comment by ATH1909** [comment]
*sigh*

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
There's more, but this is all I could muster tonight

File: code/game/objects/items/melee/misc.dm [hunk 1]
```diff
@@ -186,15 +186,24 @@
...
-	var/knockdown_time_carbon = (1.5 SECONDS) // Knockdown length for carbons.
-	var/stun_time_silicon = (5 SECONDS) // If enabled, how long do we stun silicons.
-	var/stamina_damage = 55 // Do we deal stamina damage.
-	var/affect_silicon = FALSE // Does it stun silicons.
-	var/on_sound // "On" sound, played when switching between able to stun or not.
-	var/on_stun_sound = 'sound/effects/woodhit.ogg' // Default path to sound for when we stun.
-	var/stun_animation = TRUE // Do we animate the "hit" when stunning.
-	var/on = TRUE // Are we on or off.
+	/// Default wait time until can stun again.
+	var/cooldown = 40
```
[the remark this answers is not in the document]
Comment (ATH1909) [reply code/game/objects/items/melee/misc.dm:190]: sure

File: code/game/objects/items/melee/misc.dm [hunk 2]
```diff
@@ -211,11 +220,10 @@
 	if(stamina_damage != 0)
 		offensive_notes = "\nVarious interviewed security forces report being able to beat criminals into exhaustion with only [span_warning("[round(100 / stamina_damage, 0.1)] hit\s!")]"
 
-// Description for trying to stun when still on cooldown.
 /obj/item/melee/classic_baton/proc/get_wait_description()
 	return
```
[the remark this answers is not in the document]
Comment (ATH1909) [reply code/game/objects/items/melee/misc.dm:225]: dunno, I think I might have accidentally removed it when resolving a merge conflict. I'll add it back.

File: code/game/objects/items/melee/misc.dm [hunk 3]
```diff
@@ -233,21 +242,30 @@
...
-/obj/item/melee/classic_baton/proc/additional_effects_carbon(mob/living/target, mob/living/user)
+/// Default message for trying to stun a cyborg with a baton that can't stun cyborgs.
+/obj/item/melee/classic_baton/proc/get_unga_dunga_cyborg_stun_description(mob/living/target, mob/living/user)
+	. = list()
+
+	.["visible"] = "<span class='danger'>[user] tries to knock down [target] with [src], and predictably fails!</span>" //look at this duuuuuude
+	.["local"] = "<span class='userdanger'>[target] tries to... knock you down with [src]?</span>" //look at the top of his head!
+
+	return .
+
```
Comment (LemonInTheDark) [inline code/game/objects/items/melee/misc.dm:262]: I hate this, but it's not your fault, you were failed by our past

**Pull request merged**